src/open_inwoner/utils/itertools.py

```diff
@@ -0,0 +1,24 @@
+from itertools import islice
+from math import ceil
+
+
+def batched(iterable, chunk_size: int):
+    """
+    split `iterable `in parts of `chunk_size` length
+
+    implements Python 3.12 itertools.batched()
```

Comment:
Cool, but is there a reason for not simply copying over the function definition from Python 3.12?

Also, if we do type annotations, they should be consistent I think. So either `def batched(iterable: Iterable, chunk_size: int)` or `def batched(iterable, n)`, letting the `n` speak for itself (see the docs).

Reply:
- This seems good enough. We should update python in general but that is out of scope.
- I updated the type hint, with generics
- I renamed the module to `iteration`